src/services/arweave/SimpleArweaveDataProvider.ts

```diff
@@ -98,6 +102,59 @@ export class SimpleArweaveDataProvider implements ArweaveDataProvider {
     });
   }
 
+  async validateArweaveAddress(address: string): Promise<undefined | boolean> {
+    try {
+      const targetAddress = new ArweaveTransactionID(address);
+
+      const txPromise = this._arweave.api
+        .get(`/tx/${targetAddress.toString()}`)
+        .then((res: ResponseWithData<TransactionHeaders>) =>
+          res.status === 200 ? res.data : undefined,
+        );
+
+      const balancePromise = this._arweave.api
+        .get(`/wallet/${targetAddress.toString()}/balance`)
+        .then((res: ResponseWithData<number>) =>
+          res.status === 200 ? res.data > 0 : undefined,
+        );
+
+      const gqlPromise = this._arweave.api
+        .post(`/graphql`, transactionByOwnerQuery(targetAddress))
+        .then((res: ResponseWithData<any>) =>
+          res.status === 200 ? res.data.data.transactions.edges : [],
+        );
+
+      const [isTransaction, balance, hasTransactions] = await Promise.all([
+        txPromise,
+        balancePromise,
+        gqlPromise,
+      ]);
+
+      if (hasTransactions.length || balance) {
+        return true;
+      }
+
+      if (isTransaction) {
+        const tags = tagsToObject(isTransaction.tags);
+
+        const isContract = Object.values(tags).includes('SmartWeaveContract');
+
+        throw new Error(
+          `Provided address (${targetAddress.toString()} is a ${
+            isContract ? 'Smartweave Contract' : 'transaction ID'
+          }.`,
+        );
+      }
+      // test address : ceN9pWPt4IdPWj6ujt_CCuOOHGLpKu0MMrpu9a0fJNM
+      // must be connected to a gateway that fetches L2 to perform this check
+      if (!hasTransactions || !hasTransactions.length) {
+        throw new Error(`Address has no transactions`);
+      }
```

Comment:
we should return here. i would by default `return false`

Reply:
added in [5e19895](https://github.com/ar-io/pdns-react/pull/209/commits/5e198950eab58f24ac225a45b7b6344ecbd133a1)